### ISSUE TITLE
Will workflow support loop iteration in the future?

### ISSUE THREAD

**Issue by longXboy** [langgenius/dify#3634]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

For some tasks with lengthy context, it is not suitable to send all the context to the LLM at once. Instead, the task needs to be divided into subtasks and sent to the LLM. When there are a large number of subtasks, a loop iteration is required, following the pattern: subtask -> LLM -> logical judgment -> end or continue with the next subtask.

### 2. Describe the feature you'd like to see

workflow support loop iteration：subtask -> LLM -> logical judgment -> end or continue with the next subtask

### 3. How will this feature improve your workflow or experience?

handle complex tasks

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by crazywoola** [comment]
Yes see https://docs.dify.ai/features/workflow/key-concept coming soon.
